File: bike-shop/frontend/src/services/brandService.js
import axiosClient from "./axiosConfig";

const brandService = {
  getAll: () => axiosClient.get("/brands"),
  getById: (id) => axiosClient.get(`/brands/${id}`),
  create: (data) => axiosClient.post("/brands", data),
  update: (id, data) => axiosClient.put(`/brands/${id}`, data),
  delete: (id) => axiosClient.delete(`/brands/${id}`),
};

export default brandService;

File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/BrandService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.BrandDTO;

import java.util.List;

public interface BrandService {
    List<BrandDTO> getAll();
    BrandDTO getById(Long id);
    BrandDTO create(BrandDTO dto);
    BrandDTO update(Long id, BrandDTO dto);
    void delete(Long id);
}
